packages/gotrue/lib/src/gotrue_client.dart

```diff
@@ -1128,6 +1143,59 @@ class GoTrueClient {
     _currentUser = null;
   }
 
+  void mayStartBroadcastChannel() {
+    if (const bool.fromEnvironment('dart.library.html') && _broadcastSession) {
+      // Used by the js library as well
+      final broadcastKey =
+          "sb-${Uri.parse(_url).host.split(".").first}-auth-token";
+
+      _broadcastChannel = web.getBroadcastChannel(broadcastKey);
+      _broadcastChannel?.onMessage.listen((messageEvent) {
+        final rawEvent = messageEvent['event'];
+        final event = switch (rawEvent) {
+          // This library sends the js name of the event to be comptabile with
+          // the js library, so we need to convert it back to the dart name
+          'INITIAL_SESSION' => AuthChangeEvent.initialSession,
+          'PASSWORD_RECOVERY' => AuthChangeEvent.passwordRecovery,
+          'SIGNED_IN' => AuthChangeEvent.signedIn,
+          'SIGNED_OUT' => AuthChangeEvent.signedOut,
+          'TOKEN_REFRESHED' => AuthChangeEvent.tokenRefreshed,
+          'USER_UPDATED' => AuthChangeEvent.userUpdated,
+          'MFA_CHALLENGE_VERIFIED' => AuthChangeEvent.mfaChallengeVerified,
+          // This case should never happen though
+          _ => AuthChangeEvent.values
+              .firstWhereOrNull((event) => event.name == rawEvent),
+        };
+
+        if (event != null) {
+          Session? session;
+          if (messageEvent['session'] != null) {
+            try {
+              session = Session.fromJson(messageEvent['session']);
+            } catch (e) {
+              // ignore
+              return;
+            }
+          }
+          if (session != null) {
+            _saveSession(session);
+          } else {
+            _removeSession();
+          }
+          notifyAllSubscribers(event, session: session, broadcast: false);
+        }
+      });
+    }
+  }
+
+  void dispose() {
```

Comment:
I'd add a `mustCallSuper` annotation from the [`meta` package](https://pub.dev/packages/meta)

```suggestion
  @mustCallSuper
  void dispose() {
```